src/core/classification/qgsclassificationmethod.cpp

```diff
@@ -161,6 +161,42 @@ QString QgsClassificationMethod::formatNumber( double value ) const
   }
 }
 
+void QgsClassificationMethod::addParameter( QgsProcessingParameterDefinition *definition )
+{
+  mParameters.append( definition );
+}
+
+QVariant QgsClassificationMethod::parameterValue( const QString &name ) const
+{
+  const QgsProcessingParameterDefinition *def = parameterDefinition( name );
+  if ( def )
```

Comment:
Hm.. I'm unsure on this part -- you aren't using the regular processing "parameterAs..." methods to evalute the parameter value to a variant. While this works fine with just the boolean parameter used here, I'm concerned that you'll eventually end up needing to re-implement all the complex logic which sits in the parameterAs... methods.

I think we could avoid this by creating a QgsProcessingFeedback and QgsProcessingContext object and then using the standard evaluation methods here.

Reply:
I switched to using QgsProcessingParameters::paramAsXxxxx